pkg/controller/job/backoff_utils.go

```diff
@@ -183,12 +183,18 @@ func getFinishedTime(p *v1.Pod) time.Time {
 }
 
 func getFinishTimeFromContainers(p *v1.Pod) *time.Time {
-	var finishTime *time.Time
-	for _, containerState := range p.Status.ContainerStatuses {
-		if containerState.State.Terminated == nil {
-			return nil
-		}
-		if containerState.State.Terminated.FinishedAt.Time.IsZero() {
+	finishTime := latestFinishTime(nil, p.Status.ContainerStatuses)
+	// We need to check InitContainerStatuses here also,
+	// because with the sidecar (restartable init) containers,
+	// sidecar containers will always finish later than regular containers.
+	return latestFinishTime(finishTime, p.Status.InitContainerStatuses)
```

Comment:
How about checking the feature gate here to not break the existing behavior?

Reply:
I think if a sidecar container is already created, we should always take it into consideration. Regardless of whether the featuregate of the sidecar is enabled.

And for other init containers(not sidecar). I think this won't break the existing behavior as they will always terminate before the regular container.